src/plugins/ucc_discovery_plugin.cpp

```diff
@@ -0,0 +1,345 @@
+/*
+ *  In this plugin, we implement one way for UCC dependency discovery in table columns.
+ *  In general, sort and adjacent_find are used for duplicate detection. The sort is optimized for sorted sub-vectors which are merged as a whole.
+ *  Early outs are exploited for dictionary segments.
+ */
+
+#include "ucc_discovery_plugin.hpp"
+
+#include "expression/binary_predicate_expression.hpp"
+#include "expression/expression_utils.hpp"
+#include "expression/value_expression.hpp"
+#include "hyrise.hpp"
+#include "logical_query_plan/aggregate_node.hpp"
+#include "logical_query_plan/join_node.hpp"
+#include "logical_query_plan/lqp_utils.hpp"
+#include "logical_query_plan/predicate_node.hpp"
+#include "logical_query_plan/stored_table_node.hpp"
+#include "magic_enum.hpp"
+#include "storage/table.hpp"
+#include "utils/format_duration.hpp"
+#include "utils/timer.hpp"
+
+#include "resolve_type.hpp"
+
+namespace opossum {
+
+std::string UccDiscoveryPlugin::description() const {
+  return "Unary Unique Column Combination Discovery Plugin";
+}
+
+void UccDiscoveryPlugin::start() {}
+
+void UccDiscoveryPlugin::discover_uccs() const {
```

Comment:
This function is ~100 lines long and deeply nested. The deep nesting concerns me more than the length of the function. Please simplify to improve readability.